### PR TITLE
🏗 Run Travis integration tests on Sauce Labs with 2 different commands

### PR DESCRIPTION
**Pull request by estherkim**
Runs integration tests in 2 different commands:
`gulp integration --nobuild --compiled --saucelabs --stable`
`gulp integration --nobuild --compiled --saucelabs --beta`

Does not run tests in batches. `--stable` and `--beta` only applies for sauce labs tests.

Behavior for `gulp integration --nobuild --compiled --saucelabs` remains the same.

Rename private methods from `method()` to `method_()`.

Possible fix for https://github.com/ampproject/amphtml/issues/24575

### CODE REVIEW

File: build-system/pr-check/remote-tests.js [hunk 1]
```diff
@@ -82,7 +82,12 @@ async function main() {
       buildTargets.has('FLAG_CONFIG') ||
       buildTargets.has('INTEGRATION_TEST')
     ) {
-      timedExecOrDie('gulp integration --nobuild --compiled --saucelabs');
+      timedExecOrDie(
+        'gulp integration --nobuild --compiled --saucelabs --stable'
+      );
+      timedExecOrDie(
```
Comment (rsimha) [inline build-system/pr-check/remote-tests.js]: I wonder if this should be changed to `timedExec` (without `orDie`) so beta tests cannot fail Travis irrespective of what kind of error is encountered. For example, should the "karma detected zero tests" error fail beta browser tests? WDYT?

File: build-system/tasks/karma.conf.js [hunk 2]
```diff
@@ -304,7 +304,7 @@ module.exports = {
   // TODO(rsimha): Reduce this number once keepalives are implemented by
   // karma-sauce-launcher.
   // See https://github.com/karma-runner/karma-sauce-launcher/pull/161.
-  browserDisconnectTimeout: 15 * 60 * 1000,
+  browserDisconnectTimeout: 2 * 60 * 1000,
```
Comment (rsimha) [inline build-system/tasks/karma.conf.js]: For posterity, can you add a note to the description explaining what this setting does and why it was changed?

File: build-system/tasks/runtime-test/helpers.js [hunk 3]
```diff
@@ -373,10 +431,10 @@ async function createKarmaServer(
   });
 
   karmaServer
-    .on('run_start', () => karmaRunStart())
-    .on('browsers_ready', () => karmaBrowsersReady())
-    .on('browser_start', browser => karmaBrowserStart(browser))
-    .on('browser_complete', browser => karmaBrowserComplete(browser))
+    .on('run_start', () => karmaRunStart_())
```
Comment (rsimha) [inline build-system/tasks/runtime-test/helpers.js]: While you're cleaning up this code, would this abbreviated notation work? (Will `browser` be implicitly passed in to `karmaBrowserStart_` and `karmaBrowserComplete_`?
```js
  karmaServer
    .on('run_start', karmaRunStart_)
    .on('browsers_ready', karmaBrowsersReady_)
    .on('browser_start', karmaBrowserStart_)
    .on('browser_complete', karmaBrowserComplete_)
    .on('run_complete', runCompleteFn);
```

File: build-system/tasks/karma.conf.js (hunk 2 above)
Comment (estherkim) [reply build-system/tasks/karma.conf.js]: Done.

File: build-system/pr-check/remote-tests.js (hunk 1 above)
Comment (estherkim) [reply build-system/pr-check/remote-tests.js]: Good idea, trying that.

File: build-system/tasks/runtime-test/helpers.js (hunk 3 above)
Comment (estherkim) [reply build-system/tasks/runtime-test/helpers.js]: Done

File: build-system/pr-check/remote-tests.js [hunk 4]
```diff
@@ -29,6 +29,7 @@ const {
   stopTimer,
   startSauceConnect,
   stopSauceConnect,
+  timedExec,
```
Comment (rsimha) [inline build-system/pr-check/remote-tests.js]: Minor nit for future cleanup: `timedExec` prints the wrong filename in the logs. Might need something like:

https://github.com/ampproject/amphtml/blob/f0de64b1bf8729b18e778c0801535bdc41610fa0/build-system/pr-check/remote-tests.js#L40-L41

**Comment by rsimha** [comment]
I think this change is causing the integration test Github check to error out. https://github.com/ampproject/amphtml/pull/24619/checks?check_run_id=227541301

**Pull request merged**